src/refactor_nrepl/s_expressions.clj

```diff
@@ -30,24 +30,26 @@
                        (zip/string zloc)))
           (take-while (complement nil?) (iterate zip/left zloc)))))
 
+(defn- node-at-loc?
+  "True if node encloses point defined by `loc-line` and `loc-column`."
+  [zloc ^long loc-line ^long loc-column]
+  (let [[line end-line column end-column] (->> (zip/node zloc)
+                                               meta
+                                               ((juxt :row :end-row :col :end-col))
+                                               (map (comp dec long)))]
+    (or (< line loc-line end-line)
+        (and (or (= line loc-line)
+                 (= end-line loc-line))
+             (<= column loc-column end-column)))))
+
 (defn- zip-to
-  "Move the zipper to the node at line and col"
-  [zipper ^long line ^long col]
-  (let [distance (fn [zloc]
-                   (let [node (zip/node zloc)
-                         line-beg (dec (long (:row (meta node))))
-                         line-end (dec (long (:end-row (meta node))))
-                         col-beg (dec (long (:col (meta node))))
-                         col-end (dec (long (:end-col (meta node))))]
-                     (+ (* 1000 (Math/abs (- line line-beg)))
-                        (* 100 (Math/abs (- line line-end)))
-                        (* 10 (Math/abs (- col col-beg)))
-                        (Math/abs (- col col-end)))))]
-    (reduce (fn [best zloc] (if (< (long (distance zloc)) (long (distance best)))
-                              zloc
-                              best))
-            zipper
-            (all-zlocs zipper))))
+  "Move the zipper to the node at `loc-line` and `loc-col`."
+  [zipper ^long loc-line ^long loc-column]
+  (reduce
+   (fn [best zloc]
+     (if (node-at-loc? zloc loc-line loc-column) zloc best))
```

Comment:
I really wish I had written a small comment explaining why I took this approach.  I can only guess that it was because the alternative wasn't feasible.  I tried looking through the code-base for the the source commit but it doesn't hold any further clues.

Reply:
struggle to see how this change could cause performance issues.
- `zip-to` uses the same kind of `reduce` (on the same seq etc) as the previous impl, only the pred changed
- the pred retrieves the same info, eg calls `zip/node` and `meta` on nodes than the previous impl and then just compares numbers while the previous impl did some more calculations to get `distance`